tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/rag/EmailParser.java

```diff
@@ -0,0 +1,62 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+package com.linagora.tmail.mailet.rag;
+
+import java.util.regex.Pattern;
+
+public class EmailParser {
```

Comment:
I'm curious about the level of performance here and would be happy to see JMH (Java Micro-bencmark Harness) benchmarks on real world intput....

CF exemples in https://github.com/apache/james-mime4j/blob/master/benchmark/src/main/java/org/apache/james/mime4j/JMHFieldBench.java